src/beliefpropagation/beliefpropagation.jl

```diff
@@ -172,3 +212,24 @@ function approx_network_region(
 
   return environment_tn ⊗ verts_tn
 end
+
+"""
+Return a custom edge order for how to update all BP message tensors on a general undirected graph. 
+On a tree this will yield a sequence which only needs to be performed once. Based on forest covers and depth first searches amongst the forests.
+"""
+function belief_propagation_edge_sequence(
+  g::NamedGraph; root_vertex=NamedGraphs.default_root_vertex
+)
+  @assert !is_directed(g)
+  forests = NamedGraphs.forest_cover(g)
+  edges = NamedEdge[]
+  for forest in forests
+    trees = NamedGraph[forest[vs] for vs in connected_components(forest)]
```

Comment:
```suggestion
    trees = [forest[vs] for vs in connected_components(forest)]
```
Julia will figure out the element type of the collection automatically.